### PR TITLE
Updated RBAC and CIS-Benchmark version to fix RKE1 and RKE2 hardened cluster checks

### PR DESCRIPTION
**Pull request by rayandas**
Related issue: https://github.com/rancher/rancher/issues/40818 and https://github.com/rancher/rancher/issues/40817 (changes for this issue is dependent on https://github.com/rancher/security-scan/pull/114)

- [x] TODO: Need to bump security-scan version when the above security-scan PR gets merged and tagged. 

This PR adds the necessary resources in RBAC in order to successfully run the RKE1/RKE2 hardened  check `5.1.5` and RKE1 hardened check `5.7.4`. These changes fixes the permission errors which we got in the mentioned issues mentioned above.

### CODE REVIEW

File: charts/rancher-cis-benchmark/4.0.0-rc4/templates/rbac.yaml [hunk 1]
```diff
@@ -62,10 +71,45 @@ rules:
...
   - "nodes"
   - "pods"
+  - "serviceaccounts"
+  - "services"
+  - "replicationcontrollers"
   verbs:
   - "get"
   - "list"
   - "watch"
+  - "patch"
```
Comment (macedogm) [inline charts/rancher-cis-benchmark/4.0.0-rc4/templates/rbac.yaml]: Is the `patch` verb needed?

File: packages/rancher-cis-benchmark/charts/templates/rbac.yaml [hunk 2]
```diff
@@ -62,10 +71,45 @@ rules:
...
   - "nodes"
   - "pods"
+  - "serviceaccounts"
+  - "services"
+  - "replicationcontrollers"
   verbs:
   - "get"
   - "list"
   - "watch"
+  - "patch"
```
Comment (macedogm) [inline packages/rancher-cis-benchmark/charts/templates/rbac.yaml]: Same comment as above.

File: charts/rancher-cis-benchmark/4.0.0-rc4/templates/rbac.yaml (hunk 1 above)
Comment (rayandas) [reply charts/rancher-cis-benchmark/4.0.0-rc4/templates/rbac.yaml]: Ah! right. No we don't need it. Removed and updated the PR.

File: packages/rancher-cis-benchmark/charts/templates/rbac.yaml (hunk 2 above)
Comment (rayandas) [reply packages/rancher-cis-benchmark/charts/templates/rbac.yaml]: Removed it and updated the PR

**Review by macedogm — DISMISSED** [review]

**Review by macedogm — DISMISSED** [review]

**Comment by rayandas** [comment]
The CI is failing for unknown reason. It's not able to pull `https://charts.appscode.com/stable/kubed/kubed-v0.13.2.tgz`
I tested it locally by doing `make prepare PACKAGE=epinio`, it's getting pulled successfully. Below is the output:
```
> make prepare PACKAGE=epinio                                 
INFO[0000] Pulling https://github.com/dexidp/helm-charts/releases/download/dex-0.12.1/dex-0.12.1.tgz from upstream into charts 
INFO[0001] Loading dependencies for chart               
INFO[0001] Applying changes from generated-changes      
INFO[0001] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0001] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/values.yaml.patch 
INFO[0001] Adding: charts/templates/psp.yaml            
INFO[0001] Pulling https://github.com/epinio/helm-charts/releases/download/epinio-1.6.2/epinio-1.6.2.tgz from upstream into charts 
INFO[0003] Loading dependencies for chart               
INFO[0003] Found chart options for dex in generated-changes/dependencies/dex/dependency.yaml 
INFO[0003] Found chart options for minio in generated-changes/dependencies/minio/dependency.yaml 
INFO[0003] Found chart options for kubed in generated-changes/dependencies/kubed/dependency.yaml 
INFO[0003] Found chart options for epinio-ui in generated-changes/dependencies/epinio-ui/dependency.yaml 
INFO[0003] Pulling https://charts.min.io/helm-releases/minio-5.0.4.tgz from upstream into charts 
INFO[0004] Loading dependencies for chart               
INFO[0004] Applying changes from generated-changes      
INFO[0004] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0004] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0004] Applying: generated-changes/patch/templates/post-job.yaml.patch 
INFO[0004] Applying: generated-changes/patch/templates/statefulset.yaml.patch 
INFO[0004] Applying: generated-changes/patch/values.yaml.patch 
INFO[0004] Adding: charts/templates/psp.yaml            
INFO[0004] Package epinio/dex seems to be already prepared, skipping prepare 
INFO[0004] Pulling https://github.com/epinio/helm-charts/releases/download/epinio-ui-1.5.3/epinio-ui-1.5.3.tgz from upstream into charts 
INFO[0005] Loading dependencies for chart               
INFO[0005] Applying changes from generated-changes      
INFO[0005] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0005] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0005] Applying: generated-changes/patch/templates/server.yaml.patch 
INFO[0005] Applying: generated-changes/patch/values.yaml.patch 
INFO[0005] Adding: charts/templates/security.yaml       
INFO[0005] Pulling https://charts.appscode.com/stable/kubed/kubed-v0.13.2.tgz from upstream into charts 
INFO[0006] Loading dependencies for chart               
INFO[0006] Applying changes from generated-changes      
INFO[0006] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0006] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0006] Applying: generated-changes/patch/values.yaml.patch 
INFO[0006] Adding: charts/templates/psp.yaml            
INFO[0006] Updating chart metadata with dependencies    
WARN[0006] Detected 'apiVersion:v2' within Chart.yaml; these types of charts require additional testing 
INFO[0006] Applying changes from generated-changes      
INFO[0006] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0006] Applying: generated-changes/patch/questions.yml.patch 
INFO[0006] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0006] Applying: generated-changes/patch/templates/default-app-chart.yaml.patch 
INFO[0006] Applying: generated-changes/patch/templates/server.yaml.patch 
INFO[0006] Applying: generated-changes/patch/values.yaml.patch 
INFO[0006] Adding: charts/app-readme.md                 
INFO[0006] Adding: charts/assets/epinio-application-0.1.24.tgz 
INFO[0006] Adding: charts/templates/assets.yaml         
INFO[0006] Adding: charts/templates/psp.yaml            
INFO[0006] Adding: charts/templates/validate-cert-manager-crd.yaml 
INFO[0006] Adding: charts/templates/validate-psp-install.yaml 
INFO[0006] Removing: charts/templates/default-app-namespace.yaml 
INFO[0006] Removing: charts/templates/helm-controller.yaml 
INFO[0006] Loading dependencies for chart               
INFO[0006] Copying CRDs from charts/crds to charts-crd/templates 
INFO[0006] Deleting charts/crds/app-crd.yaml            
INFO[0006] Deleting charts/crds/appcharts-crd.yaml      
INFO[0006] Deleting charts/crds/service-crd.yaml        
INFO[0006] Adding templates/validate-install-crd.yaml to main chart 
INFO[0006] Pulling https://github.com/epinio/helm-charts/releases/download/epinio-ui-1.5.3/epinio-ui-1.5.3.tgz from upstream into charts 
INFO[0007] Loading dependencies for chart               
INFO[0007] Applying changes from generated-changes      
INFO[0007] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0007] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0007] Applying: generated-changes/patch/templates/server.yaml.patch 
INFO[0007] Applying: generated-changes/patch/values.yaml.patch 
INFO[0007] Adding: charts/templates/security.yaml       
INFO[0007] Pulling https://charts.appscode.com/stable/kubed/kubed-v0.13.2.tgz from upstream into charts 
INFO[0007] Loading dependencies for chart               
INFO[0007] Applying changes from generated-changes      
INFO[0007] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0007] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0007] Applying: generated-changes/patch/values.yaml.patch 
INFO[0007] Adding: charts/templates/psp.yaml            
INFO[0007] Pulling https://charts.min.io/helm-releases/minio-5.0.4.tgz from upstream into charts 
INFO[0007] Loading dependencies for chart               
INFO[0007] Applying changes from generated-changes      
INFO[0007] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0007] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0007] Applying: generated-changes/patch/templates/post-job.yaml.patch 
INFO[0007] Applying: generated-changes/patch/templates/statefulset.yaml.patch 
INFO[0007] Applying: generated-changes/patch/values.yaml.patch 
INFO[0007] Adding: charts/templates/psp.yaml
```

**Review by macedogm — DISMISSED** [review]

**Comment by macedogm** [comment]
@rayandas last time that I had such similar issue, the problem was in the tgz generated charts file. Have you tried decompressing them manually (locally) and see if they are inside the right directories and have all the expected files?

**Comment by rayandas** [comment]
Yeah I’ve checked that. Also tried installing the chart using helm. It got installed successfully. 
Seems like some CI issue. Though it’s fixed now. 
Could you please approve it once again? @macedogm

**Review by macedogm — APPROVED** [review]

**Pull request merged**